etest/tests.py

```diff
@@ -137,13 +137,14 @@ def run(self):
 
             tag_name = str(self.commands.index(command))
 
+            commit_name = self.name.replace("=", "").replace("[", "").replace("]", "")
             docker.container.commit(
-                container_name,
-                repository=self.name,
+                container,
+                repository=commit_name,
                 tag=tag_name,
             )
 
-            image_name = self.name + ":" + tag_name
+            image_name = commit_name + ":" + tag_name
```

Comment:
I'd argue that commit should return the image name but we can figure that out later if that doesn't make sense.